data/data/agent/files/usr/local/bin/load-config-iso.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -e
+
+AGENT_CONFIG_ARCHIVE_FILE="config.gz"
+AGENT_CONFIG_MOUNT="/media/config-image"
+CLUSTER_IMAGE_SET="/etc/assisted/manifests/cluster-image-set.yaml"
+AGENT_CONFIG_MOUNT="/media/config_image"
+
+copy_archive_contents() {
+    tmpdir=$(mktemp --tmpdir -d "config-image--XXXXXXXXXX")
+    cp ${AGENT_CONFIG_MOUNT}/${AGENT_CONFIG_ARCHIVE_FILE} "${tmpdir}"
+    gunzip -f "${tmpdir}"/${AGENT_CONFIG_ARCHIVE_FILE}
+    unzipped_file=$(echo "${tmpdir}/${AGENT_CONFIG_ARCHIVE_FILE}" | cut -d'.' -f1)
+    filelist=$(cpio --list < "${unzipped_file}")
+    echo -e "List of files to copy: ${filelist}"
+
+    # Get the releaseImage in the archive and verify it matches the current cluster-image-set
+    release_image=$(grep releaseImage ${CLUSTER_IMAGE_SET} | sed -n -e 's/^.*releaseImage: //p')
+    arch_image_set=$(cpio -icv --to-stdout ${CLUSTER_IMAGE_SET} < "${unzipped_file}")
+    if [[ $(< "${CLUSTER_IMAGE_SET}") != "${arch_image_set}" ]]; then
```

Comment:
Can we have this validation according to ClusterImageSet's name (i.e. the version in binary)?
That way, we could support also openshift-install binaries downloaded from: https://mirror.openshift.com/pub/openshift-v4/amd64/clients/ocp.
As the releaseImage url is different than the binary in release (registry.ci.openshift.org/origin/release vs quay.io/openshift-release-dev/ocp-release)

Reply:
I think this is probably not what we want do, as we want the releaseImage of the config-image to exactly match the releaseImage of the unconfigured igntion. Wdyt @zaneb ?

Reply:
We really want it to exactly match, otherwise we can never change anything in the installer.
What we need is to provide a way for users to get the installer from the release image that is preinstalled on the servers (or even run it on one of those servers).